### ISSUE TITLE
pink material!

### ISSUE THREAD

**Issue by nebis-software** [itsFulcrum/Unity-URP-Hair-Shader#1]
Hi,
   I am trying to use your nice shader with a hair model of mine , i am using unity 2022.4
From the 3d artist i got normal map, opacity, diffuse, transparency and base color. 
Applying them to your shader i always get a pink material.... do you have any idea / hints to share? I really like the youtube demo and I would like to test it

Thanks!
Luigi

**Comment by itsFulcrum** [comment]
Hard to say without more information. What errors are you seeing?

**Comment by nebis-software** [comment]
No errors, it is just pink... Where can I see errors?
My impression is that the shader was created with a previous version of URP
(13 maybe?) and Unity 2022 has URP 14

Happy to send a test project if this helps
Many thanks!!
Luigi

**Comment by itsFulcrum** [comment]
I don't have time at the moment to test it as I am on vacation.  Maybe in 2 weeks.    Make sure all files are in the same folder.  If you select the main shader file it should give you errors in the inspector view.    It's possible that the newer version of urp broke it but I'd say that's rather unlikely, I'm not aware of any mayor changes.

**Comment by nebis-software** [comment]
Hi,
     here is the error I am getting (from Inspector -> compile and show code)
PS: my target platform is Android, does it make a difference?

![image](https://github.com/itsFulcrum/Unity-URP-Hair-Shader/assets/132885313/ba4a0faf-a5ca-47e1-ada7-98bd3e3d4d17)

Thanks!

**Comment by xrr-233** [comment]
Chaging unity to the corresponding version is ok for me. My previous version was too high.

**Comment by Jaimi** [comment]
To fix this in 2022.3+, find this line in Hairlighting.hlsl:

`       uint meshRenderingLayers = GetMeshRenderingLightLayer();`
       
And change it to look like this:
       
```
  #if UNITY_VERSION < 202220
       uint meshRenderingLayers = GetMeshRenderingLightLayer();
  #else
       uint meshRenderingLayers = GetMeshRenderingLayer();
  #endif     
```

**Comment by itsFulcrum** [comment]
> To fix this in 2022.3+, find this line in Hairlighting.hlsl:
> 
> ` uint meshRenderingLayers = GetMeshRenderingLightLayer();`
> 
> And change it to look like this:
> 
> ```
>   #if UNITY_VERSION < 202220
>        uint meshRenderingLayers = GetMeshRenderingLightLayer();
>   #else
>        uint meshRenderingLayers = GetMeshRenderingLayer();
>   #endif     
> ```

Closing issue since unity aparently updated their naming for no reason.
